src/main/java/com/nike/cerberus/store/ConfigStore.java

```diff
@@ -186,6 +186,32 @@ public void storeCert(CertificateInformation certificateInformation,
         return getDecryptedEnvironmentData().getCertificateData();
     }
 
+    /**
+     * Deletes a set of cert and key files by certificate name
+     * @param certificateName the name of the cert file bundle to delete
+     */
+    public void deleteCertificate(String certificateName) {
+        initEncryptedConfigStoreService();
+
+        encryptedConfigStoreService.deleteAllKeysOnPartialPath("certificates/" + certificateName);
+        synchronized (cerberusEnvironmentDataLock) {
```

Comment:
Not sure what this synchronized is doing?  Is this multi-threaded?

Reply:
Just following the pattern already established, but yes it is not guaranteed that this is running in a single thread.

Reply:
As in, it is not guaranteed today?  Or in the future?

Reply:
I would have to dig through the source code, there a few things that happen multi-threaded. This pattern is already used all over this class. We can make a ticket to address further if you want.

Reply:
No biggy, just trying to understand.  If it is an established pattern, it was probably someone trying to future proof it.